homeassistant/components/tradfri/base_class.py

```diff
@@ -61,17 +71,17 @@ def _async_start_observe(self, exc=None):
             _LOGGER.warning("Observation failed, trying again", exc_info=err)
             self._async_start_observe()
 
-    async def async_added_to_hass(self):
+    async def async_added_to_hass(self) -> None:
         """Start thread when added to hass."""
         self._async_start_observe()
 
     @callback
-    def _observe_update(self, device):
+    def _observe_update(self, device: Command) -> None:
```

Comment:
Why a device is a command? Naming seems to be a bit odd...

Reply:
hmmm did you notice that Command is used throughout the module. Without a typed library, it is sometimes a bit of guessing what it the best fit.

Reply:
you might also notice (if you look in the library) that Device do not exist only DeviceInfo.

the library is part of the home-assistant family, so we can change it, which was why I somewhere else asked for an opinion from Martin.

Reply:
Please have a look at pytradfri/__init__:
```
def get_devices(self):
        """
        Return the devices linked to the gateway.

        Returns a Command.
        """
```
this is called in __init__ of this integration to get a list of devices, so it might be odd, but it is correct.

Reply:
We pass the command to get the devices to the api which returns a list of devices.

https://github.com/home-assistant/core/blob/a57d7717a88d7a53ff8c8e45fabc29bd3e2721e5/homeassistant/components/tradfri/__init__.py#L111-L112

Reply:
Complicated library :smile: